dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DataverseSessionConfigListener.java

```diff
@@ -0,0 +1,32 @@
+package edu.harvard.iq.dataverse;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+import javax.servlet.SessionCookieConfig;
+
+@WebListener
+public class DataverseSessionConfigListener implements ServletContextListener {
+
+    @Override
+    public void contextInitialized(ServletContextEvent sce) {
+        SessionCookieConfig sessionCookieConfig = sce.getServletContext().getSessionCookieConfig();
+        String cookieName = System.getenv("COOKIE_NAME");
+        if (cookieName != null) {
+            sessionCookieConfig.setName(cookieName);
+        }
```

Comment:
If it is possible to use `SystemServiceBean` to get the properties instead of system environment then I would go with `SystemServiceBean`. If injecting services in `@WebListener` is not possible then current solution is fine IMO.